python/ray/experimental/sgd/model.py

```diff
@@ -7,16 +7,61 @@ class Model(object):
     """Your class must implement this interface to be used with Ray SGD.
 
     This supports any form of input pipeline: it is up to you to define it
-    using TensorFlow. The only requirements are that the loss and optimizer
-    attributes must be defined.
-
+    using TensorFlow.
     For an example implementation, see tfbench/test_model.py
-
-    Attributes:
-        loss (tf.Tensor): Loss function to minimize.
-        optimizer (tf.train.Optimizer): Optimizer to use to minimize the loss.
     """
 
+    def build(self):
+        """Build customized tensorflow model
+
+        Users should setup variables such as loss or in get_loss function
+        """
+        raise NotImplementedError('build of %s is not implemented' % self.__class__.__name__)
+
+    def get_loss(self):
+        """Return loss of the model
+
+        Returns:
+            loss
+        """
+        raise NotImplementedError('get_loss of %s is not implemented' % self.__class__.__name__)
+
+    #TODO support complex way of updating gradient, e.g. using different optimizers
+    def get_optimizer(self):
+        """Return optimizer for the model
+
+        Returns:
+            optimizer
+        """
+        raise NotImplementedError('get_optimizer of %s is not implemented' \
+                % self.__class__.__name__)
+
+    def get_metrics(self):
+        """Return metrics of the model
+
+        Returns:
+            metrics(dict): e.g. {'accuray': accuracy_tensor}
```

Comment:
```suggestion
            metrics(dict): e.g. {'accuracy': accuracy_tensor}
```

Reply:
Metrics are not currently reported right? Do you plan to add it?

Reply:
No, they are not. It is used to replace methods such as `test_accuracy`. But I have not figured out a way to replace them. These methods may need different feed dict, but currently model has only one `get_feed_dict` method. Maybe `get_metrcis` can just return numpy arrays like `test_accuracy` using `eval` of tensor.